Add serveral observability metrics

qwes5s5 opened the pull request:
fastdeploy:cache_config_info 类型为 Gauge，记录了当前节点缓存设置(CacheConfig)相关的信息，当引擎进行初始化时进行记录，内含有：
* block_size: KV缓存中每个块（block）的大小，以token为单位。
* bytes_per_block: 每个KV缓存块所占用的字节数。
* bytes_per_layer_per_block: 每个块在单层模型中占用的字节数。
* cache_dtype: 缓存中存储键值（KV）对的数据类型。
* cache_queue_port: 用于缓存队列通信的端口号。
* cache_transfer_protocol: 缓存数据传输使用的协议，例如 ipc（进程间通信）。
* dec_token_num: 解码阶段的令牌数量。
* each_token_cache_space: 每个令牌在KV缓存中占用的空间大小。
* enable_chunked_prefill: 是否启用分块预填充。
* enable_hierarchical_cache: 是否启用分层缓存。
* enable_prefix_caching: 是否启用前缀缓存。
* enable_ssd_cache: 是否启用SSD固态硬盘缓存。
* enc_dec_block_num: 编码器-解码器模型的块数量。
* gpu_memory_utilization: 允许使用的GPU内存占总GPU内存的比例。
* kv_cache_ratio: KV缓存占用的内存比例。
* max_block_num_per_seq: 每个序列（请求）允许分配的最大块数量。
* model_cfg: 模型的配置对象。
* num_cpu_blocks: 分配给CPU的缓存块数量。
* num_gpu_blocks_override: 覆盖默认值后，分配给GPU的缓存块数量。
* pd_comm_port: 用于通信的端口号。
* prealloc_dec_block_slot_num_threshold: 预先分配解码块槽的阈值。
* prefill_kvcache_block_num: 预填充阶段使用的KV缓存块数量。
* rdma_comm_ports: 用于RDMA（远程直接数据存取）通信的端口号。
* swap_space: 交换空间的大小，用于处理内存不足的情况。
* total_block_num: KV缓存的总块数。

示例：
```
# HELP fastdeploy:cache_config_info Information of the engine's CacheConfig
# TYPE fastdeploy:cache_config_info gauge
fastdeploy:cache_config_info{block_size="64",bytes_per_block="589824",bytes_per_layer_per_block="32768",
cache_dtype="bfloat16",cache_queue_port="8003",cache_transfer_protocol="ipc",dec_token_num="128",
each_token_cache_space="9216",enable_chunked_prefill="False",enable_hierarchical_cache="False",
enable_prefix_caching="False",enable_ssd_cache="False",enc_dec_block_num="2",gpu_memory_utilization="0.9",
kv_cache_ratio="0.75",max_block_num_per_seq="128",model_cfg="<fastdeploy.config.ModelConfig object at 0x7f74a6b413f0>",
num_cpu_blocks="0",num_gpu_blocks_override="1000",pd_comm_port="None",prealloc_dec_block_slot_num_threshold="5",
prefill_kvcache_block_num="750",rdma_comm_ports="None",swap_space="None",total_block_num="1000"} 1.0
```
fastdeploy:available_batch_size，类型为Gauge，记录了当前节点可用批量大小，表示还能接受多少新的请求，在资源分配和回收时记录
```
# HELP fastdeploy:available_batch_size Number of how many new requests the system can still accept.
# TYPE fastdeploy:available_batch_size gauge
fastdeploy:available_batch_size 128.0
```
fastdeploy:hit_req_rate，类型为Gauge，记录了Request级别前缀缓存命中率，在CacheMetrics更新命中metrics时进行记录
```
# HELP fastdeploy:hit_req_rate Request-level prefix cache hit rate
# TYPE fastdeploy:hit_req_rate gauge
fastdeploy:hit_req_rate 0.5
```
fastdeploy:hit_token_rate，类型为Gauge，记录了Token级别的前缀缓存命中率，在CacheMetrics更新命中metrics时进行记录
```
# HELP fastdeploy:hit_token_rate Token-level prefix cache hit rate
# TYPE fastdeploy:hit_token_rate gauge
fastdeploy:hit_token_rate 0.5
```
fastdeploy:cpu_hit_token_rate，类型为Gauge，记录了Token级别cpu前缀缓存命中率，在CacheMetrics更新命中metrics时进行记录
```
# HELP fastdeploy:cpu_hit_token_rate Token-level CPU prefix cache hit rate
# TYPE fastdeploy:cpu_hit_token_rate gauge
fastdeploy:cpu_hit_token_rate 0.0
```
fastdeploy:gpu_hit_token_rate，类型为Gauge，记录了Token级别GPU前缀缓存命中率，在CacheMetrics更新命中metrics时进行记录
```
# HELP fastdeploy:gpu_hit_token_rate Token-level GPU prefix cache hit rate
# TYPE fastdeploy:gpu_hit_token_rate gauge
fastdeploy:gpu_hit_token_rate 0.5
```

Commit:
Add several observability metrics

Commit:
[wenxin-tools-584] 【可观测性】支持查看本节点的并发数、剩余block_size、排队请求数等信息

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/PaddlePaddle/FastDeploy?pullRequest=3762) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/PaddlePaddle/FastDeploy?pullRequest=3762) before we can accept your contribution.<br/>**13** out of **14** committers have signed the CLA.<br/><br/>:white_check_mark: ltd0924<br/>:white_check_mark: Echo-Nie<br/>:white_check_mark: co63oc<br/>:white_check_mark: ooooo-create<br/>:white_check_mark: chang-wenbin<br/>:white_check_mark: rsmallblue<br/>:white_check_mark: plusNew001<br/>:white_check_mark: carryyu<br/>:white_check_mark: lizexu123<br/>:white_check_mark: XieYunshen<br/>:white_check_mark: luukunn<br/>:white_check_mark: aquagull<br/>:white_check_mark: YuanRisheng<br/>:x: K11OntheBoat<br/><hr/>**K11OntheBoat** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/PaddlePaddle/FastDeploy?pullRequest=3762) it.</sub>

Commit:
Merge branch 'develop' into feature/add-monitoring-metrics

Commit:
add metrics unit test

Commit:
Merge branch 'develop' into feature/add-monitoring-metrics

ltd0924 committed:
[BugFix] fix scheduler invalid (#3803)

* [BugFix] fix max streaming tokens invalid

* fix scheduler bug

* fix scheduler bug

co63oc committed:
rename fused_get_rope.cu (#3752)

* rename fused_get_rope.cu

* fix

* fix typos

* fix

* fix

Echo-Nie committed:
【Hackathon 9th No.84】Supplementary Unit Test for fastdeploy/reasoning (#3570)

测试内容：测试基类的注册、获取函数功能是否正常

Co-authored-by: Tao Luo <luotao02@baidu.com>

co63oc committed:
fix w8a8.py (#3733)

co63oc committed:
fix dcu_worker.py (#3734)

ooooo-create committed:
【Hackathon 9th No.73】add unit tests for graph_opt_backend (#3609)

* test: add unit tests for graph_opt_backend

* refactor(tests): improve graph optimization test structure and readability

* fix(tests): correct CUDA graph related typos in test files

- Fix class name: TestCUDAGrpahSubgraph -> TestCUDAGraphSubgraph

* refactor(test): support attention layer and optimize graph optimization backend test to eliminate redundant baseline calculations

* remove some func call

---------

Co-authored-by: RAM <gstian5555@outlook.com>
Co-authored-by: Tao Luo <luotao02@baidu.com>

plusNew001 committed:
[XPU] FIX XPU CI BUG  (#3829)

* Add debug environment variable exports

Added debug environment variable exports for CLANG_PATH and XVLLM_PATH.

* Lock paddlepaddle-xpu version in CI script

Temporarily lock paddlepaddle-xpu version due to framework update issues.

chang-wenbin committed:
[Doc] update wint2 doc (#3819)

* update_wint2_doc

carryyu committed:
fix test_append_attention_with_output.py (#3831)

Co-authored-by: plusNew001 <95567040+plusNew001@users.noreply.github.com>

plusNew001 committed:
[XPU] Update XPU CI case (#3837)

* Add debug environment variable exports

Added debug environment variable exports for CLANG_PATH and XVLLM_PATH.

* Lock paddlepaddle-xpu version in CI script

Temporarily lock paddlepaddle-xpu version due to framework update issues.

* Update no_proxy environment variable in CI workflow

* Install lsof tool in run_ci_xpu.sh

rsmallblue committed:
qk norm for speculate decode C16 (#3637)

YuanRisheng committed:
V1 loader support ep (#3801)

lizexu123 committed:
[Code Simplification] delete cum_offsets_out (#3815)

* fix

* fix

Jiang-Jia-Jun commented on fastdeploy/output/token_processor.py:
```diff
@@ -282,6 +282,7 @@ def _recycle_resources(self, task_id, index, task, result=None, is_prefill=False
         main_process_metrics.batch_size.set(
             self.resource_manager.max_num_seqs - self.resource_manager.available_batch()
         )
+        main_process_metrics.available_batch_size.set(self.resource_manager.available_batch())
```
avaiable_batch多次更新，在此处更新一次即可？

Jiang-Jia-Jun commented:
另外可以更新一下docs/zh/online_serving/metrics.md（包括对应英文文档）

aquagull committed:
[Feature] `ernie4_5_vl_moe` support huggingface safetensor loading (#3750)

* update

* update

* update in tp

* add todo

* update

---------

Co-authored-by: aquagull <hongyuh@qq.com>

luukunn committed:
add reasoning parser plugin (#3811)

* add reasoning parser plugin

* fix finish reason

XieYunshen committed:
adjust some metrics and md file

Commit:
Revert "Merge branch 'develop' into feature/add-monitoring-metrics"

This reverts commit 07c62346021da2e096dbe3e5fc5ad9cac991bc2f, reversing
changes made to f90f8d6a76107467c47730156d3699738a13b538.

Commit:
adjust some metrics and md files

Commit:
Merge branch 'develop' into feature/add-monitoring-metrics
